src/main/java/io/kokuwa/keycloak/metrics/store/MicrometerEventStoreProviderFactory.java

```diff
@@ -0,0 +1,36 @@
+package io.kokuwa.keycloak.metrics.store;
+
+import io.kokuwa.keycloak.metrics.CommunityProfiles;
+import org.keycloak.Config;
+import org.keycloak.connections.jpa.JpaConnectionProvider;
+import org.keycloak.events.EventStoreProvider;
+import org.keycloak.events.jpa.JpaEventStoreProviderFactory;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.provider.EnvironmentDependentProviderFactory;
+
+
+public class MicrometerEventStoreProviderFactory extends JpaEventStoreProviderFactory implements EnvironmentDependentProviderFactory {
+
+    public static final String ID = "jpa2"; //override
+
+    @Override
+    public EventStoreProvider create(KeycloakSession session) {
+        JpaConnectionProvider connection = session.getProvider(JpaConnectionProvider.class);
+        return new MicrometerEventStoreProvider(session, connection.getEntityManager());
+    }
+
+    @Override
+    public String getId() {
+        return ID;
+    }
+
+    @Override
+    public boolean isSupported() {
+        return CommunityProfiles.isEventsMetricsEnabled();
+    }
+
+    @Override
+    public boolean isSupported(Config.Scope config) {
```

Comment:
And this is not enabling/disabling it? Add some logging to see what happens.

Reply:
Done